hooks/checkout

```diff
@@ -122,20 +131,27 @@ checkout() {
     # being advertised. In that case fetch the branch instead.
     elif [[ "${exit_code}" -ne 0 ]]; then
       exit_code=0
-      timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v --no-tags origin "${BUILDKITE_BRANCH}" || exit_code=$?
+      echo "---------- timeout "${GIT_REMOTE_TIMEOUT}" GIT_TRACE=1 GIT_CURL_VERBOSE=1 git fetch -v --no-tags origin "${BUILDKITE_BRANCH}" || exit_code=$? ----------" >> "${log}"
+      (GIT_TRACE=1 GIT_CURL_VERBOSE=1 timeout "${GIT_REMOTE_TIMEOUT}"git fetch -v --no-tags origin "${BUILDKITE_BRANCH}" || exit_code=$?) &>> "${log}"
       check_timeout_exit_code "${exit_code}" "${GIT_REMOTE_TIMEOUT_EXIT_CODE}"
     fi
     # If the commit doesn't exist the ref that was pointing to it might have
     # been force pushed in the meantime. Exit with ESTALE to signify the stale
     # branch reference in that case.
     exit_code=0
-    git checkout -f "${BUILDKITE_COMMIT}" || exit_code=$?
+    echo "---------- GIT_TRACE=1 GIT_CURL_VERBOSE=1 git checkout -f ${BUILDKITE_COMMIT} || exit_code=$? ----------" >> "${log}"
+    (GIT_TRACE=1 GIT_CURL_VERBOSE=1 git checkout -f "${BUILDKITE_COMMIT}" || exit_code=$?) &>> "${log}"
     if [[ "${exit_code}" -eq 128 ]]; then
       exit 116
     elif [[ "${exit_code}" -ne 0 ]]; then
       exit "${exit_code}"
     fi
   fi
+
+  if is_running_on_buildkite; then
+    buildkite-agent artifact upload "${log}"
+  fi
+  rm -rf "${log}"
```

Comment:
this should be a trap, to avoid leaving files when the checkout fails.
Also uploading the file should also be part of the trap, otherwise we will only upload logs of successful checkouts.

Reply:
good point, will update